prettyprinter/src/Prettyprinter/Internal.hs

```diff
@@ -452,6 +455,15 @@ instance Pretty a => Pretty (Maybe a) where
 -- Manually use @'hardline'@ if you /definitely/ want newlines.
 instance Pretty Text where pretty = vsep . map unsafeTextWithoutNewlines . T.splitOn "\n"
 
+-- | Convert text to Doc. Must not contain newlines. 
+-- Useful when dealing with wide characters and emojis
```

Comment:
The documentation should clarify why and how this function is useful for dealing with wide characters and emojis.